R/feature.align.R

```diff
@@ -34,22 +92,22 @@ feature.align <- function(features,
     if (number_of_samples > 1) {
         values <- get_feature_values(features, rt_colname)
         mz_values <- values$mz
-        chr <- values$rt
-        lab <- values$sample_id
+        rt <- values$rt
+        sample_id <- values$sample_id
         
-        o <- order(mz_values, chr)
-        mz_values <- mz_values[o]
-        chr <- chr[o]
-        lab <- lab[o]
+        # sort all values by m/z, if equal by rt
+        ordering <- order(mz_values, rt)
+        mz_values <- mz_values[ordering]
+        rt <- rt[ordering]
+        sample_id <- sample_id[ordering]
         
         # find relative m/z tolerance level
         if (is.na(mz_tol_relative)) {
             mz_tol_relative <- find.tol(mz_values, mz_max_diff = mz_max_diff, do.plot = do.plot)
             if (length(mz_tol_relative) == 0) {
                 mz_tol_relative <- 1e-5
-                warning(
-                    "Automatic tolerance finding failed, 10 ppm was assigned. May need to manually assign alignment mz tolerance level."
-                )
+                warning("Automatic tolerance finding failed, 10 ppm was assigned. 
+                        May need to manually assign alignment mz tolerance level.")
```

Comment:
This is something that we should address maybe: I doubt someone will see this warning given that most of our tests produce dozens of warning.

Reply:
Do you think the function should fail with an error? Or we could parametrise the whole function to use default value vs. fail with an error.

Reply:
No, I don't think an error would be an appropriate solution. I just wanted to point out that apLCMS may produce a meaningful warning, so we should make sure that the warning doesn't get lost in a bunch of deprecation warnings, etc. I think we just have to reduce the number of other warnings like deprecation, imports, and so on. Should be done with #93.